File: Practice(Beginner)/C++/Chef and String.cpp
// Source: https://www.codechef.com/problems/RECNDSTR
#include <bits/stdc++.h>
using namespace std;

string comp(string s) {
    string myLStr = "";
    string myRStr = "";

    for (int i = 1; i < s.size(); ++i) {
        myLStr += s[i];
    }
    myRStr += s[s.size()-1];
    for (int i = 0; i < s.size()-1; ++i) {
        myRStr += s[i];
    }
    myLStr += s[0];
    return (myLStr==myRStr ? "YES" : "NO");
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        string s;
        cin >> s;
        cout << comp(s) << endl;
    }
}

File: Practice(Beginner)/Python3/Chef_and_String.py
# Source: https://www.codechef.com/problems/RECNDSTR
for test in range(int(input())):
    s = input()
    left = s[1:] + s[0]
    right = s[-1] + s[:-1]
    print("YES") if(left == right) else print("NO")